ARCs/arc-0018.md

```diff
@@ -0,0 +1,211 @@
+
+An example may include some music rights where the label, the artist, and any investors have some assigned royalty percentage that should be enforced on transfer. During the sale transaction, the appropriate royalty payments should be included or the transaction must be rejected.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+
+[Royalty Policy](#royalty-policy) - The data structure that defines how royalties are dispersed
+
+[Royalty Enforcer](#royalty-enforcer) - The contract that enforces the royalty payments given the Royalty Policy and performs transfers of the assets.
+
+[Royalty Collector](#royalty-collector) - The account that collect the royalty and provides the royalty distribution logic. It could be set to a Single Signature, a Multi Signature, a Smart Signature or even another Smart Contract.
+
+[Asset Subject to Royalties](#royalty-asset) - The asset that is subject to royalty payments on transfer
+
+[Third Party Marketplace](#third-party-marketplace) - A third party marketplace may be any marketplace that implements the appropriate methods to initiate transfers.
+
+### Royalty Policy 
+
+A Royalty Share consists of an `address` that should receive a Royalty payment and a `share` representing some share of the total payment amount. The Royalty Share MUST be specified in terms of [basis points](https://en.wikipedia.org/wiki/Basis_point) of the payment amount.
+
+A Royalty Policy is specified as a Royalty Collector and associated Royalty weighting. Additionally, if any assets besides Algos are acceptable, 
+
+
+```ts
+type RoyaltyPolicy {
+    share:              number,  // The share of the payment due, specified in basis points
+    royalty_collector:  string,  // The address that should collect this share of the payment
+    allowed_assets:     number[] // The list of asset ids that are acceptable for payment
+}
+```
+
+
+### Royalty Enforcer
+
+The Royalty Enforcer is an application that controls the transfer of ASAs subject to the Royalty Policy. 
+
+This is accomplished by exposing an interface defined as a set of [ABI Methods](#abi-methods) allowing a grouped transaction call.
+
+### Royalty Collector
+
+The Royalty Collector is a generic account that could be set to a Single Signature, a Multi Signature, a Smart Signature or even to another Smart Contract. The Royalty Collector is then responsible for the royalty distribution logic, making the Royalty Enforcement Specification more general and composable.
+
+
+### Royalty Asset 
+
+The Royalty Asset is an ASA subject to royalty collection and MUST be created with the the appropriate roles and parameters. 
+
+> Because the protocol does not allow updating a role after it's been deleted, if the creator thinks she may want to modify them later, they must be set to some non-zero address.
```

Comment:
Are we talking explicitly about the clawback address here?

Reply:
clawback/freeze both I think